### ISSUE TITLE
No Pull Request comment added on push

### ISSUE THREAD

**Issue by MichaelJHaywood** [Madrapps/jacoco-report#62]
I've been having trouble trying to get the plugin to add a comment to the PR. I've enabled `debug-mode` and it successfully finds the Jacoco xml file. It outputs messages about the changes but doesn't mention trying to add a comment. The action is successful. 

```
- name: Publish code coverage results
  id: jacoco
  uses: madrapps/jacoco-report@v1.6.1
  with:
    paths: |
      jacoco/jacocoTestReport.xml
    token: ${{ github.token }}
    min-coverage-overall: 40
    min-coverage-changed-files: 60
    title: Code Coverage
    update-comment: true
    debug-mode: true
    continue-on-error: false
```

**Logs**
```
Run madrapps/jacoco-report@v1.6.1
Event is push
passEmoji: :green_apple:
failEmoji: :x:
base sha: ...
head sha: ...
reportPaths: jacoco/jacocoTestReport.xml
Resolved files: ...
file: {
...
]
project: {
...
}
skip: false
prNumber: undefined
```

I can successfully add comments myself using a separate github action so I don't believe this is permission related. 

**Adds a comment successfully**
```
- name: PR Comment Test
  uses: mshick/add-pr-comment@v2
  with:
    message: |
      **Test**
```

Am I using this incorrectly? Should it fail if it doesn't add a comment?

**Comment by MichaelJHaywood** [comment]
The problem seems to be here: 

**action.ts**
```
if (prNumber != null && !skip) {
```

If the prNumber is undefined we don't add a comment. In the case of a "push" there is no prNumber set. 

I've managed to get this working by getting the PR number associated with the commit.

```
case 'push':
  base = github.context.payload.before
  head = github.context.payload.after

  const prs: any[] = (await client.rest.repos.listPullRequestsAssociatedWithCommit({
    commit_sha: github.context.sha,
    owner: github.context.repo.owner,
    repo: github.context.repo.repo,
  })).data

  if(prs.length > 0){
     prNumber = prs[0].number
  }

  break
```

Overall this means pushing changes will not update the code coverage comment without this change (or similar).

**Comment by patrykosak** [comment]
@thsaravana 
Can you reopen issue and apply this fix on master branch?

**Comment by thsaravana** [comment]
I am already making some changes in 1.7 in this area. So I will test this as well and will fix this in 1.7 properly.

**Comment by vincent-fuchs** [comment]
isn't the workaround also to trigger the workflow not on `push` event, but on `pull_request` `opened` / `reopened` / `synchronize` ? I think a push in a branch for which there's a PR always generates a `synchronize`. and you would not want the action to run on a `push` if there's no PR associated to the branch, right ?

**Comment by guus-bloemsma** [comment]
> isn't the workaround also to trigger the workflow not on `push` event, but on `pull_request` `opened` / `reopened` / `synchronize` ? I think a push in a branch for which there's a PR always generates a `synchronize`. and you would not want the action to run on a `push` if there's no PR associated to the branch, right ?

The problem with that is that you always want to build and test on push, at least in main. Adding a comment to an associated PR is an optional addition. I wouldn't want to rebuild/retest everything just for the coverage comment.

**Comment by sheepdreamofandroids** [comment]
Is there still any progress on this?

**Comment by jameshreaver** [comment]
Same problem here, my workflows are triggered by `push` and I don't want to have to re-build everything in a workflow with a  `pull_request` trigger. Would it be possible to either allow a PR number as an input or to try to get the PR number on `push` workflows too? It should be possible

**Comment by thsaravana** [comment]
We will release this in `v1.7.2`.

- There will now be an input field `pr-number` where we can input "any" Pull Request number and the Comment will be added to that, [eg. here](https://github.com/thsaravana/jacoco-android-playground/pull/15/commits/7b10c394552f84215a0f7575c2dec9eb25248e99)
- If `pr-number` is not provided, then we will figure out the PR Number from the environment. For `pull_request` event this happens already, and for `push` event (and more events we are planning to add like `workflow_run`) this will be calculated from the commit_sha. [eg. here](https://github.com/thsaravana/jacoco-android-playground/pull/15)

PS: In addition to this, we have also added "Workflow Summary", so for "push" to `main` branch, you don't really need to go to the associated PR to get the coverage report, but can get the same information in the action workflow summary. [eg. here ](https://github.com/thsaravana/jacoco-android-playground/actions/runs/10928733611)

All of this will be released in the next one: `v1.7.2`

**Comment by thsaravana** [comment]
This is now available in [v1.7.2-alpha](https://github.com/Madrapps/jacoco-report/releases/tag/v1.7.2-alpha). We will release v1.7.2 in the coming weeks.
